pyupgrade/_plugins/io_open.py

```diff
@@ -32,3 +33,8 @@ def visit_Call(
             node.func.attr == 'open'
     ):
         yield ast_to_offset(node.func), _replace_io_open
+
+        # apply open() upgrades
+        node.func = node.func.value
+        node.func.id = 'open'
```

Comment:
the ast should be treated as immutable -- this is a hack I can't accept

Reply:
Faire enough.
If you have any idea/suggestion/guidance on how to implement it without the hack, I will happily give it another try, otherwise feel free to close the PR.

Reply:
my guess is that the `open` check would just need to learn about `io.open` and also probably the `io.open` rewriter would need to keep the same offsets when rewriting (if it doesn't already)